testing/unittests/esdltests.cpp

```diff
@@ -0,0 +1,801 @@
+              </if>
+            </when>
+            <when test="$AssertLevel2C">
+               <assert test="not($AssertLevel2C)" code="23" message="'AssertLevel2C'"/>
+            </when>
+            <otherwise>
+               <set-value target="test"  select="'auth success'"/>
+               <set-value target="Row/Name/Last" select="'XXX'"/>
+               <set-value target="Row/Name/Last" select="'POE'"/>
+               <append-to-value target="Row/AppendTo"  select="'This'"/>
+               <append-to-value target="Row/AppendTo"  select="'One'"/>
+               <append-to-value target="Row/AppendTo"  select="'String'"/>
+               <add-value target="Row/Name/Aliases/Alias"  select="'moe'"/>
+               <add-value target="Row/Name/Aliases/Alias"  select="'poe'"/>
+               <add-value target="Row/Name/Aliases/Alias"  select="'roe'"/>
+            </otherwise>
+         </choose>
+      </otherwise>
+   </choose>
+</CustomRequestTransform>
+)!!";
+
+
+static const char *target_config = "<method queryname='EchoPersonInfo'/>";
+
+class ESDLTests : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE( ESDLTests );
+        CPPUNIT_TEST(testEsdlTransformScript);
+        CPPUNIT_TEST(testEsdlTransformScriptNoPrefix);
+        CPPUNIT_TEST(testEsdlTransformForEach);
+        CPPUNIT_TEST(testEsdlTransformLegacy);
+        CPPUNIT_TEST(testEsdlTransformIgnoreScriptErrors);
+        CPPUNIT_TEST(testEsdlTransformTargetXpathErrors);
+        CPPUNIT_TEST(testEsdlTransformFailStrict);
+        CPPUNIT_TEST(testEsdlTransformScriptVarParam);
+        CPPUNIT_TEST(testEsdlTransformFailLevel1A);
+        CPPUNIT_TEST(testEsdlTransformFailLevel1B);
+        CPPUNIT_TEST(testEsdlTransformFailLevel1C);
+        CPPUNIT_TEST(testEsdlTransformFailLevel2A);
+        CPPUNIT_TEST(testEsdlTransformFailLevel2B);
+        CPPUNIT_TEST(testEsdlTransformFailLevel2C);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+    ESDLTests(){}
+
+    inline const char *queryTestName(IPropertyTree *cfg)
+    {
+        const char *testname = cfg->queryProp("Transform/Param[@name='testcase']/@value");
+        if (!testname)
+            return "unknown";
+        return testname;
+    }
+    void runTest(const char *runscript, const char *xml, const char *config, const char *result, int code)
+    {
+        StringBuffer request(xml);
+        Owned<IPropertyTree> cfg = createPTreeFromXMLString(config);
+        const char *testname = queryTestName(cfg);
+
+        try
+        {
+            Owned<IPropertyTree> target = createPTreeFromXMLString(target_config);
+            Owned<IPropertyTree> script = createPTreeFromXMLString(runscript);
+            Owned<IEsdlCustomTransform> tf = createEsdlCustomTransform(*script, nullptr);
+
+            Owned<IEspContext> ctx = createEspContext(nullptr);//createHttpSecureContext(m_request.get()));
+            tf->processTransform(ctx, target, "EsdlExample", "EchoPersonInfo", "EchoPersonInfoRequest", request, cfg);
```

Comment:
Maybe beyond the scope of a unit test, but since we're not calling processServiceAndMethodTransforms() there is no test of the strictParams treatment of User Properties and the other values added via addInputValue() vs. those added via addVariable().

Reply:
It actually is calling processServiceAndMethodTransforms() see the test named testEsdlTransformFailStrict that specifically tests strictParam.